java/ql/lib/semmle/code/java/frameworks/apache/Codec.qll

```diff
@@ -0,0 +1,135 @@
+        "org.apache.commons.codec.language.bm;Languages$LanguageSet;true;restrictTo;(LanguageSet);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Languages$SomeLanguages;false;getLanguages;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Languages;true;getLanguages;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;PhoneticEngine;true;getLang;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Rule$Phoneme;false;Phoneme;(CharSequence,LanguageSet);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.language.bm;Rule$Phoneme;false;Phoneme;(CharSequence,LanguageSet);;Argument[1];Argument[-1];taint;",
+        "org.apache.commons.codec.language.bm;Rule$Phoneme;false;Phoneme;(Phoneme,Phoneme,LanguageSet);;Argument[2];Argument[-1];taint;",
+        "org.apache.commons.codec.language.bm;Rule$Phoneme;false;append;(CharSequence);;Argument[-1];ReturnValue;value;",
+        "org.apache.commons.codec.language.bm;Rule$Phoneme;false;getLanguages;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Rule$Phoneme;false;getPhonemeText;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Rule$Phoneme;false;mergeWithLanguage;(LanguageSet);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.language.bm;Rule$Phoneme;false;mergeWithLanguage;(LanguageSet);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Rule$Phoneme;false;toString;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Rule$PhonemeExpr;true;getPhonemes;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Rule$PhonemeList;false;PhonemeList;(List);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.language.bm;Rule;true;Rule;(String,String,String,PhonemeExpr);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.language.bm;Rule;true;Rule;(String,String,String,PhonemeExpr);;Argument[3];Argument[-1];taint;",
+        "org.apache.commons.codec.language.bm;Rule;true;getLContext;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Rule;true;getPattern;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Rule;true;getPhoneme;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Rule;true;getRContext;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language;Caverphone;true;caverphone;(String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.language;DoubleMetaphone$DoubleMetaphoneResult;true;getAlternate;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language;DoubleMetaphone$DoubleMetaphoneResult;true;getPrimary;();;Argument[-1];ReturnValue;taint;",
```

Comment:
The methods that could taint [`DoubleMetaphoneResult`](https://commons.apache.org/proper/commons-codec/apidocs/org/apache/commons/codec/language/DoubleMetaphone.DoubleMetaphoneResult.html) (`append*`) should be added too to complete these models.